src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -816,29 +814,28 @@ public unsafe void CopyTo(Array array, int index)
                 Vector128<byte> lowerShuffleMask_CopyToBoolArray = Vector128.Create(0, 0x01010101_01010101).AsByte();
                 Vector128<byte> upperShuffleMask_CopyToBoolArray = Vector128.Create(0x02020202_02020202, 0x03030303_03030303).AsByte();
 
-                if (Avx2.IsSupported)
+                if (Vector256.IsHardwareAccelerated)
                 {
                     Vector256<byte> shuffleMask = Vector256.Create(lowerShuffleMask_CopyToBoolArray, upperShuffleMask_CopyToBoolArray);
                     Vector256<byte> bitMask = Vector256.Create(0x80402010_08040201).AsByte();
                     Vector256<byte> ones = Vector256.Create((byte)1);
 
-                    fixed (bool* destination = &boolArray[index])
+                    ref byte destination = ref Unsafe.As<bool, byte>(ref Unsafe.Add<bool>(ref MemoryMarshal.GetArrayDataReference<bool>(boolArray), index));
+
+                    for (; (i + Vector256ByteCount) <= (uint)m_length; i += Vector256ByteCount)
```

Comment:
```suggestion
                    for (; i <= (uint)m_length - Vector256ByteCount; i += Vector256ByteCount)
```
Otherwise `i + Vector256ByteCount` needs to be evaluated on each iteration?

Reply:
Using `Vector256<byte>.Count` is also better than some `Vector256ByteCount`. The JIT special cases the vector `.Count` properties in a few cases today and it may result in better codegen in some scenarios.